server/routes/dev_place/utils.py

```diff
@@ -90,3 +98,374 @@ def get_place_type_with_parent_places_links(dcid: str) -> str:
+
+    # If not found, check if the string matches the default locale (e.g., '@en')
+    if string_with_locale.endswith(f"@{DEFAULT_LOCALE}"):
+      # If a default locale string is found, store it to return later if needed
+      default_i18n_string = string_with_locale.rsplit("@", 1)[0]
+
+  # Return the string with the default locale if no match was found for the desired locale
+  return default_i18n_string
+
+
+def fetch_place(place_dcid: str, locale: str = DEFAULT_LOCALE) -> Place:
+  """
+  Fetches a single Place object for the given place DCID and locale.
+
+  Args:
+      place_dcid (str): The unique identifier (DCID) for the place.
+      locale (str): The desired locale for the place name (default: 'en').
+
+  Returns:
+      Place: A Place object with the requested locale's name and other details.
+  """
+  places = fetch_places([place_dcid], locale)
+  return places[0]
+
+
+def fetch_places(place_dcids: List[str], locale=DEFAULT_LOCALE) -> List[Place]:
+  """
+  Fetches a list of Place objects for the given list of place DCIDs and locale.
+
+  Args:
+      place_dcids (List[str]): A list of unique identifiers (DCIDs) for places.
+      locale (str): The desired locale for the place names (default: 'en').
+
+  Returns:
+      List[Place]: A list of Place objects with names in the specified locale.
+  """
+  place_types = fetch.property_values(place_dcids, 'typeOf')
+  place_name = fetch.property_values(place_dcids, 'name')
+
+  # Only fetch names with locale-specific tags if the desired locale is non-english
+  place_name_with_languages = {}
+  if locale != DEFAULT_LOCALE:
+    place_name_with_languages = fetch.property_values(place_dcids,
+                                                      'nameWithLanguage')
+
+  places = []
+  for place_dcid in place_dcids:
+    place_name_with_languages_strs = place_name_with_languages.get(
+        place_dcid, [])
+    name_with_locale = select_string_with_locale(place_name_with_languages_strs,
+                                                 locale=locale)
+    default_name = next(iter(place_name.get(place_dcid, []))) or ''
+
+    # Use the name with locale if available, otherwise fall back to the default ('en') name
+    name = name_with_locale or default_name
+    places.append(
+        Place(dcid=place_dcid, name=name, types=place_types[place_dcid]))
+    place_types[place_dcid]
+  return places
+
+
+def chart_config_to_overview_charts(chart_config):
+  """
+  Converts the given chart configuration into a list of Chart objects for API responses.
+
+  Args:
+      chart_config (List[Dict]): A list of chart configuration dictionaries, 
+                                  each containing metadata like 'category', 'statsVars', 'title', etc.
+
+  Returns:
+      List[Chart]: A list of Chart objects created from the chart configuration.
+  """
+  charts = []
+  for page_config_item in chart_config:
+    denominator = next(iter(page_config_item.get("denominator", [])), None)
```

Comment:
if i'm reading this right, this gets the first denominator from the list.

could you make sure it works in cases like [this](https://github.com/datacommonsorg/website/blob/master/server/config/chart_config/equity.json#L54)
the sv and denominator should come from the same index where the denominator.size() > 1

Reply:
Good catch! fixed